### ISSUE TITLE
Add Projection Tool

### ISSUE THREAD

**Issue by TimPurdum** [dymaptic/GeoBlazor#36]
Several consumers requested the ability to alter the spatial projection of geometries. The [Projection](https://developers.arcgis.com/javascript/latest/api-reference/esri-geometry-projection.html) tool handles these conversions.

1. Add a TypeScript file to call and manage `@arcgis/core/geometry/projection`.
2. Add a C# class that exposes all public methods and returns .NET-consumable Geometries or other types.
3. Add documentation for this tool.
4. Add or expand a sample page (Optional, if we can come up with a good demo of this?)

**Comment by morehavoc** [comment]
This sample should be doable using the new Projection Tools/functions (instead of the custom method):

https://blazor.dymaptic.com/source-code/projection

**Comment by morehavoc** [comment]
@TimPurdum Oops! I was thinking that was using Project, but it's just using Geodesic Buffer.

I have implemented this in Blazor-WASM though and it worked fine, no issues with lag that I saw. So maybe this needs two implementations? A version for Blazor-Server that shows how to implement a custom JS function (something that a user might need to do) and a version for Blazor-WASM that shows how to do it in c#. (this is really a separate gissue from this one)

Maybe for showing off the Projection tool, we do something like show a list of XY points that are in a State Plane or UTM zone and project those to WGS84 and add them as graphics. Not as cool as the buffer, but would show some actual code doing a projection.
